File: examples/v2/stack-config/python/__main__.py
import pulumi
import pulumi_pulumiservice.v2 as ps_v2

config = pulumi.Config()
service_org = config.get("serviceOrg") or "service-provider-test-org"
project_name = config.get("projectName") or "v2-stack-config-example"
stack_name = config.get("stackName") or "dev"
hook_url = config.get("hookUrl") or "https://example.invalid/hooks/example"
env_ref = config.get("envRef") or "organization/credentials"

parent_stack = ps_v2.Stack(
    "parentStack",
    org_name=service_org,
    project_name=project_name,
    stack_name=stack_name,
)

ps_v2.StackConfig(
    "config",
    org_name=service_org,
    project_name=parent_stack.project_name,
    stack_name=parent_stack.stack_name,
    environment=env_ref,
)

ps_v2.StackWebhook(
    "hook",
    organization_name=service_org,
    project_name=parent_stack.project_name,
    stack_name=parent_stack.stack_name,
    name="v2-stackhook",
    display_name="Stack hook example",
    payload_url=hook_url,
    active=True,
    format="pulumi",
)

pulumi.export("stack", parent_stack.id)

File: examples/v2/stack-config/go/main.go
package main

import (
	stacks "github.com/pulumi/pulumi-pulumiservice/sdk/go/pulumiservice/v2/stacks"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi/config"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		cfg := config.New(ctx, "")
		serviceOrg := cfg.Get("serviceOrg")
		if serviceOrg == "" {
			serviceOrg = "service-provider-test-org"
		}
		projectName := cfg.Get("projectName")
		if projectName == "" {
			projectName = "v2-stack-config-example"
		}
		stackName := cfg.Get("stackName")
		if stackName == "" {
			stackName = "dev"
		}
		hookUrl := cfg.Get("hookUrl")
		if hookUrl == "" {
			hookUrl = "https://example.invalid/hooks/example"
		}
		envRef := cfg.Get("envRef")
		if envRef == "" {
			envRef = "organization/credentials"
		}

		parentStack, err := stacks.NewStack(ctx, "parentStack", &stacks.StackArgs{
			OrgName:     pulumi.String(serviceOrg),
			ProjectName: pulumi.String(projectName),
			StackName:   pulumi.String(stackName),
		})
		if err != nil {
			return err
		}

		if _, err := stacks.NewConfig(ctx, "config", &stacks.ConfigArgs{
			OrgName:     pulumi.String(serviceOrg),
			ProjectName: parentStack.ProjectName,
			StackName:   parentStack.StackName,
			Environment: pulumi.String(envRef),
		}); err != nil {
			return err
		}

		if _, err := stacks.NewWebhook(ctx, "hook", &stacks.WebhookArgs{
			OrganizationName: pulumi.String(serviceOrg),
			ProjectName:      parentStack.ProjectName,
			StackName:        parentStack.StackName,
			Name:             pulumi.String("v2-stackhook"),
			DisplayName:      pulumi.String("Stack hook example"),
			PayloadUrl:       pulumi.String(hookUrl),
			Active:           pulumi.Bool(true),
			Format:           pulumi.String("pulumi"),
		}); err != nil {
			return err
		}

		ctx.Export("stack", parentStack.ID())
		return nil
	})
}
